test/unit/component/parser/json/BpmnJsonParser.label.test.ts

```diff
@@ -16,52 +16,48 @@
 import { parseJsonAndExpectOnlyEdges, parseJsonAndExpectOnlyFlowNodes } from './JsonTestUtils';
 import each from 'jest-each';
 import { TProcess } from '../../../../../src/model/bpmn/json/baseElement/rootElement/rootElement';
-
-describe('parse bpmn as json for label font', () => {
-  each([
-    ['exclusiveGateway'],
-    ['inclusiveGateway'],
-    ['parallelGateway'],
-    ['task'],
-    ['userTask'],
-    ['serviceTask'],
-    ['receiveTask'],
-    ['sendTask'],
-    ['manualTask'],
-    ['scriptTask'],
-    ['callActivity'],
-    ['subProcess'],
-    ['textAnnotation'],
-    // TODO: To uncomment when we support complex gateway
-    //['complexGateway'],
-    ['businessRuleTask'],
-  ]).it("should convert as Shape without Label, when a BPMNShape (attached to %s & with empty BPMNLabel) is an attribute (as object) of 'BPMNPlane' (as object)", sourceKind => {
-    const json = {
-      definitions: {
-        targetNamespace: '',
-        process: {
-          id: 'Process_1',
-        },
-        BPMNDiagram: {
-          id: 'BpmnDiagram_1',
-          BPMNPlane: {
-            id: 'BpmnPlane_1',
-            BPMNShape: {
-              id: 'shape_source_id_0',
-              bpmnElement: 'source_id_0',
-              Bounds: { x: 362, y: 232, width: 36, height: 45 },
-              BPMNLabel: '',
+import { ShapeBpmnElementKind } from '../../../../../src/model/bpmn/internal/shape';
+import { ShapeUtil } from '../../../../../src/bpmn-visualization';
+
+export const shapeBpmnElementKindForLabelTests = Object.values(ShapeBpmnElementKind)
```

Comment:
I prefer to keep the explicit list ```
 ['exclusiveGateway'],
    ['inclusiveGateway'],
    ['parallelGateway'],
    ['task'],
    ['userTask'],
    ['serviceTask'],
    ['receiveTask'],
    ['sendTask'],
    ['manualTask'],
    ['scriptTask'],
    ['callActivity'],
    ['subProcess'],
    ['textAnnotation'],
    // TODO: To uncomment when we support complex gateway
    //['complexGateway'],
    ['businessRuleTask'],
```, because with the refactoring, we don't test that we have the right text in the enums. If we change the value of the enum, the tests will work, but if we keep the explicit list, they fail.

Reply:
The values are tested with the parsing of each BPMN element: the value in the enum MUST match the xml node name (and so the json attribute). So this use case is tested and it should better be.

From `ProcessConverter.buildProcessInnerElements`
```typescript
    // flow nodes
    ShapeUtil.flowNodeKinds()
      .filter(kind => kind != ShapeBpmnElementKind.EVENT_BOUNDARY)
      .forEach(kind => this.buildFlowNodeBpmnElements(processId, process[kind], kind));
```

I introduced the generated list because
- lot of enum values were missing
- we should have documented to manually update this list (which is boring for the contributor)

Notice that the explicit list was duplicated in 3 tests, this should have least been refactored to keep the list at a single place.

Reply:
In the XML tests, we just verify if the XML file corresponds to the JSON model.

Here, we want to verify if the internal model is correctly build from the JSON model, and so, if the JSON model correspond to the enum (with the implementation code).

I've always heard that the test parameters should be decoloured from the implementation code.

Reply:
About the tests, I meant the gobal `BpmnParser` unit test (from xml to internal model, that doesn't cover all shape kind)s and the mxgraph model integration test (from xml to mxgraph model, that covers all shape kinds) that detect an issue in the enum values.

Here we want to exhaustively test all the enum values that describe all supported BPMN elements and I don't want to manually list the values in the test at the price of extra documentation and the risk of missing elements (the enum has no logic and is just a data structure, so I wouldn't consider it as implementation code here - just a domain representation).

Let's talk directly to converge on this topic 😸